server/game/GameActions/ConditionalAction.js

```diff
@@ -0,0 +1,42 @@
+const GameAction = require('./GameAction');
+
+class ConditionalAction extends GameAction {
+    setup() {
+        super.setup();
+        this.effectMsg = this.effect || 'conditionally do something';
```

Comment:
I don't think you should be using this.effect here

Reply:
I renamed it to `message`. It is safe to use there as it is set in constructor